meta/language.LL1

```diff
@@ -0,0 +1,181 @@
+integer     = /-?([1-9][0-9]*|0[Xx][0-9A-Fa-f]+|0[Oo][0-7]*|0)([ui](8|16|32|64)|uz)?/
+float       = /-?(([0-9]+\.[0-9]*|[0-9]*\.[0-9]+)([Ee][+-]?[0-9]+)?|[0-9]+[Ee][+-]?[0-9]+)(f(64|32))?/
+            | /-?([1-9][0-9]*|0)f(64|32)/
+identifier  = /_?[A-Za-z][0-9A-Z_a-z-]*/
+
+String      = '"' {EscapedChar} '"'
+ByteLiteral = "b'" EscapedChar "'"
+CChar       = "'" EscapedChar "'"
+
+EscapedChar = /([^\\]|\\.)/
+
+boolean_op = /[<>]|[<>=!]=/
+
+op  = /[+-*\/%|^&]/
+    | /[&|+-*\/]=/
+    | boolean_op
+    | /\?\?=?/ 
+lop = /++|--?|[*~]|/
+    | "not"
+    | "&raw"
+rop = /++|--/
+    | ("[" Expression "]")
+
+Constant         = integer
+                 | float
+                 | String
+                 | ByteLiteral
+                 | CChar
+                 | "None"
+                 | "true" | "false"
+
+BuiltinType      = /f(32|64)/
+                 | /[iu](8|16|32|64)/
+                 | "usize"
+                 | /c_(int|char)/
+                 | "bool"
+                 | "String"
+                 | "void"
+
+TypeExpression   = BuiltinType ["?"]
+                 | {identifier [GenericSpecifier] "::" } identifier [GenericSpecifier] ["?"]
+                 | "[" TypeExpression "]" ["?"]
+                 | "[" TypeExpression ":" TypeExpression "]" ["?"]
+                 | "{ TypeExpression "}" ["?"]
+                 | "raw" TypeExpression  ["?"]
+                 | "weak" TypeExpression "?"
+
+GenericSpecifier = "<" TypeExpression {"," TypeExpression} ">"
+
+ArrayLiteral     = "[" [ArrayItem [ArrayItems]] "]"
+                 | "[" Expression ";" integer "]"
+ArrayItems       = "," ArrayItem [ArrayItems]
+ArrayItem        = Expression
+
+MapLiteral       = "[:]"
+                 | "[" MapItem [MapItems] "]"
+MapItems         = "," MapItem [MapItems]
+MapItem          = Expression ":" Expression
+
+SetLiteral       = "{" [SetItem [SetItems]] "}"
+SetItems         = "," SetItem [SetItems]
+SetItem          = Expression
+
+Range            = Expression ".." Expression
+
+MatchExpression  = "match" Expression "{" {MatchCase} "}"
+MatchCase        = MatchKey "=>" Statement
+MatchKey         = MatchKey "|" MatchKey
+                 | {identifier "::"} identifier ["(" {identifier [":" identifier]} ")"]
+                 | Constant
+                 | "else"
+
+IsExpression     = Expression "is" TypeExpression
+CastExpression   = Expression /as[!?]/ TypeExpression
+
+Expression       = Expression op Expression
+                 | lop Expression
+                 | Expression rop
+                 | "(" Expression ")"
+                 | Constant
+                 | ArrayLiteral
+                 | MapLiteral
+                 | SetLiteral
+                 | Range
+                 | MatchExpression
+                 | IsExpression
+                 | CastExpression
+                 | identifier {"::" identifier}
+                 | [Expression] /!?\./ identifier
+                 | Expression "(" ArgumentList ")"
```

Comment:
I feel there are some missing `[GenericSpecifier]`s missing here
@alimpfard can you confirm that any identifier here can take a GenericSpecifier?

Reply:
Only the call target can atm
```
Expression ["<" {Typename} ">"] "(" ... ")"
```

Reply:
So something like `my_enum<i64>::myvalue` is not allowed currently?
Same for any of
```
MyClass<T>::static_function(...)
MyClass::static_function<T>(...)
generic_constant<T>
```
My guess would be that the GenericSpecifier belongs to the identifier and not to the Expression

Reply:
Yep, that's currently not valid.